### PR TITLE
CORTX-22807: Fix issues for iostat, dstat and blktrace collections

### PR DESCRIPTION
**Pull request by rahul27kumar**
Signed-off-by: Rahul Kumar <rahul.kumar@seagate.com>

## Problem Statement

Their is an issues while stats collection for iostat, dstat and blktrace. Fixes some minor issues for respective source code.

## Design

-   stats collections were not collecting properly for iostat, dstat and blktrace. 

> **Following are the checklists for the author. Make sure all the items are considered. Write NA in front if any point is not applicable.**

## Coding

-   [x] Coding conventions are followed and code is consistent
-   [x] No new Codacy Issues are added
-   [x] Version is updated in the respective VERSION file, format - `<tool>_v<major>.<minor>.<patch>`

## Documentation

-   [ ] Changes done to readme file / Quick Start Guide
-   [ ] Docstrings are added to new functions/ files with Summary, Args and Return params

## Testing

-   [x] New/Affected tests are executed on Latest Build
-   [x] Attach test execution logs
-   [x] No regression introduced

## Pull Request

-   [x] JIRA number/GitHub Issue added to PR, format - `<ID>: <Title>`
-   [x] Only one commit is present, format -
    ```txt
    commit 43e060bb2a12ddf8f9f944356b8c19a7512a5442
    Author: Rahul Kumar <rahul.kumar@seagate.com>
    Date:   Fri Sep 2 08:23:02 2022 -0600

    CORTX-22807: Fix issues for iostat, dstat and blktrace collections

    Signed-off-by: Rahul Kumar <rahul.kumar@seagate.com>

    ```
-   [x] Commit is sign-offed

## Review Checklist

-   [x] PR is self reviewed
-   [x] JIRA ticket state/status is updated
-   [x] Check if the DoD, description is clear and explained

## Impact Analysis

  (Checklist for Author/Reviewer/GateKeeper)

-   [ ] Are there any changes in any common function?

  If yes,

-   [ ] Relative/ dependent function calls are updated
-   [x] Executed all affected tests

> Note: If checkboxes are not working, write Y/N in front of the bullets. Thanks for your contribution and Happy Coding!

### CODE REVIEW

**Comment by rahul27kumar** [comment]
[workload.log](https://github.com/Seagate/seagate-tools/files/9493192/workload.log)
Logs for successful execution of perfline for iostats
Results for iostats collection on http://ssc-vm-g3-rhev4-3136.colo.seagate.com:8005/report/b232c41a-f8ee-4b43-930b-56b5df2fb1fe

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh [hunk 1]
```diff
@@ -41,49 +41,49 @@ MD_DISKS=$(cat "$DISKS_MAP" | grep 'MD:' | sed 's/MD://')
...
         dm=$(realpath "$d" | xargs basename)
         disks_dm="$disks_dm $dm"
         echo "MD $d $dm" >> disks.mapping
     done
 
-     iostat-cli --fig-size 20,20 --data iostat.log \
-           --disks $disks_dm \
-           --fig-output iostat.aggregated.png plot
+    iostat-cli --fig-size 20,20 --data iostat.log \
+               --disks ${disks_dm} \
```
Comment (rkothiya) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:66]: Please give double quote to this

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh [hunk 2]
```diff
@@ -108,22 +108,23 @@ function blktrace_service_stop()
...
          for graph in io tput latency queue_depth iops; do
-             iowatcher -t "$d".dump -o "$d".$graph.svg -O $graph
+             iowatcher -t "$d".dump -o "$d".${graph}.svg -O ${graph}
          done
      done
 
-     dumps=$(ls -1 | grep "dump" | awk '{print "-t "$1}' | tr '\n' ' ')
-     iowatcher "$dumps" -o node."$(hostname -s)".aggregated.svg
+     dumps=$(ls -1 | grep "dump" | awk '{print "-t "$1}' | tr '\n' ' ' | sed 's/ *$//g')
+     iowatcher ${dumps} -o node."$(hostname -s)".aggregated.svg
```
Comment (rkothiya) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:116]: Please give double quote to dumps.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh [hunk 3]
```diff
@@ -108,22 +108,23 @@ function blktrace_service_stop()
...
      for graph in io tput latency queue_depth iops; do
-         iowatcher "$dumps" -o node."$(hostname -s)".$graph.svg -O $graph
+         iowatcher ${dumps} -o node."$(hostname -s)".${graph}.svg -O ${graph}
```
Comment (rkothiya) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:118]: Please give double quote to dumps

File: performance/PerfLine/VERSION [hunk 4]
```diff
@@ -1 +1,2 @@
-VERSION perfline_v0.4.2
+VERSION perfline_v0.4.3
+
```
Comment (sampadap03) [inline performance/PerfLine/VERSION:2]: remove an extra line from here

**Review by sampadap03 — CHANGES_REQUESTED** [review]
small change please do

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh (hunk 1 above)
Comment (rahul27kumar) [reply performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:66]: This command will fail if we use "${disks_dm}". The commands will expect for a list of disk not a string type value.
Double quotes convert it  to string. So, We should ignore codacy issues related to this file.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh (hunk 2 above)
Comment (rahul27kumar) [reply performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:116]: This command will fail if we use "${dumps}". The commands will expect for a list of disk not a string type value.
Double quotes convert it to string. So, We should ignore codacy issues related to this file.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh (hunk 3 above)
Comment (rahul27kumar) [reply performance/PerfLine/roles/perfline_setup/files/wrapper/stat/stop_stats_service.sh:118]: same as above comment

**Comment by rahul27kumar** [comment]
Successful Tested Perfline execution: http://ssc-vm-g3-rhev4-3136.colo.seagate.com:8005/report/e7f189be-fdde-4437-901d-6caf95bc3e0a

**Review by rkothiya — APPROVED** [review]
LGTM

**Review by sampadap03 — APPROVED** [review]

**Comment by sampadap03** [comment]
merging by bypassing codacy checks with the above reasons

**Pull request merged**